presto-main/src/main/java/com/facebook/presto/cost/CostCalculatorUsingExchanges.java

```diff
@@ -119,6 +121,12 @@ public PlanNodeCostEstimate visitGroupReference(GroupReference node, Void contex
             throw new UnsupportedOperationException();
         }
 
+        @Override
+        public PlanNodeCostEstimate visitAssignUniqueId(AssignUniqueId node, Void context)
+        {
+            return cpuCost(getStats(node.getSource()).getOutputRowCount() * BigintType.BIGINT.getFixedSize());
```

Comment:
nit: `getStats(node)` should be sufficient (instead of `getStats(node.getSource())`) -- provided that we have stats rule for `AssignUniqueId`, which we need to have anyway.